### PR TITLE
Fix regrid time function to add year yr frequency and remove resetting calendars

### LINKED ISSUE #339: Bug in regrid time
Hi,

I think I've found a pair of bugs in the regrid time preprocessor.
 
Firstly, it isn't set up to re-grid annual data (no big deal - easy fix). Just add these lines at line 531 of `_time.py`. 
```
    elif frequency == 'year':
        cube.coord('time').cells = [
            datetime.datetime(t.year, 7, 1, 0, 0, 0) for t in time_c
        ]
```
Secondly, the order of operations is incorrect. At the moment, it changes the time units in line 501, then loads the time afterwards in line 507. This means that the time data is loaded using `gregorian` units, from a netcdf file which might not be in gregorian units!

Ie: This line:
https://github.com/ESMValGroup/ESMValCore/blob/11c53fa21e39301cba26ce941a233baad51b39de/esmvalcore/preprocessor/_time.py#L507

should be after this line:
https://github.com/ESMValGroup/ESMValCore/blob/11c53fa21e39301cba26ce941a233baad51b39de/esmvalcore/preprocessor/_time.py#L501

not before!

### PR DESCRIPTION
**Pull request by valeriupredoi**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/CONTRIBUTING.md): yadayada, I did, I bloody co-wrote that!

**Tasks**

-   [x] [Create an issue] Created #339 by @ledm  
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Add unit tests
-   [x] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [x] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #339 

@ledm here you go, bud! Cheers for raising that, I did remove completely the calendar resetting in `regrid_tim` since that was total bollocks and was actuallt resetting calendars and changing the time axis in a nasty way; no need to worry about cubes with different calendars since after the CMOR checks all cubes will have the same calendar :beer:

### CODE REVIEW

File: esmvalcore/preprocessor/_time.py [hunk 1]
```diff
@@ -497,15 +497,13 @@ def regrid_time(cube, frequency):
...
-    cube.coord('time').units = cf_units.Unit(
-        cube.coord('time').units.origin,
-        calendar='gregorian',
-    )
-
     # standardize time points
     time_c = [cell.point for cell in cube.coord('time').cells()]
-    if frequency == 'mon':
+    if frequency == 'yr':
+        cube.coord('time').cells = [
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_time.py]:
```suggestion
        cells = [
```
Don't replace the [cube.coord('time').cells](https://scitools.org.uk/iris/docs/v2.2.1/iris/iris/coords.html#iris.coords.Coord.cells) function with an array. Just use `cells` or some other name for your array.

**Comment by valeriupredoi** [comment]
@ledm PING :ping_pong:

**Comment by valeriupredoi** [comment]
tested it, can I get an approve for the PR please @mattiarighi (Lee is sleeping)

**Comment by valeriupredoi** [comment]
really need an approve here folks @mattiarighi @ledm @schlunma and anyone else pls

**Comment by mattiarighi** [comment]
I can't test right now since DKRZ is down.
@ledm can you confirm it solves your issue?

**Comment by ledm** [comment]
I honestly can't remember where I encountered this bug. October feels like a long time ago. The code looks reasonable, but I'm unable to figure out where the problem was or test it right now.

**Comment by mattiarighi** [comment]
Will test once finished with #420.

**Comment by valeriupredoi** [comment]
@ledm it's the bug that you reported and I told you off and then I actually realized it was indeed a bug and we opened two pr's to fix it only my pr was better than yours (no milkshakes) :grin:

**Comment by mattiarighi** [comment]
Looks to good to me, although I cannot think of a specific test case for this.

**Comment by valeriupredoi** [comment]
I tested it! @mattiarighi - "looks too good"? :rofl:

**Comment by valeriupredoi** [comment]
this merge didnt push a whole lot of refs

**Pull request merged**